### ISSUE TITLE
Allow users to bring their own version of typescript

### ISSUE THREAD

**Issue by whitneyit** [ivogabe/gulp-typescript#63]
Instead of specifying typescript as a dependency, it could be moved to a peerDependency like shown below.

```diff
 {
   "dependencies": {
-    "typescript": "1.4.1"
   },
+  "peerDependencies": {
+    "typescript": "*"
+  }
 }
```

This would allow a team to install their own version of typescript without the gulp-typescript plugin dictating the version for them.

```json
{
   "dependencies": {
    "typescript": "1.4.1",
    "gulp-typescript": "^2.3.0"
  }
}
```
  
Any thoughts?

**Comment by ivogabe** [comment]
This would probably be pretty complex, since different API versions don't behave the same. See for example #54. For future releases the changes will probably be very small, since the compiler API is now marked stable.

Do you want this feature for working with older versions or with newer versions (for example the current master)?

**Comment by whitneyit** [comment]
Purely just an idea. I was thinking it would ease the barrier to entry for
teams with an existing project on a different version than what is
specified in package.json

**Comment by alexeagle** [comment]
Now that Angular2 is in TypeScript, but the tsc version required is a preview of unreleased 1.5, there will probably be more requests for this.

**Comment by ivogabe** [comment]
I've implemented it in #76. It wasn't very easy because of the breaking changes between the API versions, for example `filename` has been renamed to `fileName`. It should support TS 1.4 and the current master now, but I expect there will be some issues with api changes coming.

It's also possible to use a custom fork of typescript. That fork has to match the api of 1.4 or the master. I'm not aware of such fork yet, `jsx-typescript` for example doesn't have the same api as 1.4 or the master (but something in between), so it doesn't work (yet?).

**Comment by ivogabe** [comment]
Note: as of 2.6.0, you need to add the custom version of typescript to the settings object, like this:
```javascript
[...].pipe(ts({
    typescript: require('typescript')
});
```
See #84
